rcl/include/rcl/arguments.h

```diff
@@ -0,0 +1,187 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__ARGUMENTS_H_
+#define RCL__ARGUMENTS_H_
+
+#include "rcl/allocator.h"
+#include "rcl/macros.h"
+#include "rcl/types.h"
+#include "rcl/visibility_control.h"
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+struct rcl_arguments_impl_t;
+
+/// Hold output of parsing command line arguments.
+typedef struct rcl_arguments_t
+{
+  /// Private implementation pointer.
+  struct rcl_arguments_impl_t * impl;
+} rcl_arguments_t;
+
+/// Return a rcl_node_t struct with members initialized to `NULL`.
+RCL_PUBLIC
+RCL_WARN_UNUSED
+rcl_arguments_t
+rcl_get_zero_initialized_arguments(void);
+
+/// Parse command line arguments into a structure usable by code.
+/**
+ * If an argument does not appear to be a valid ROS argument then it is skipped and parsing
+ * continues with the next argument in `argv`.
+ * \sa rcl_arguments_get_count_unparsed()
+ * \sa rcl_arguments_get_unparsed()
+ *
+ * Successfully parsed remap rules are stored in the order they were given in `argv`.
+ * If given arguments `{"__ns:=/foo", "__ns:=/bar"}` then the namespace used by nodes in this
+ * process will be `/foo` and not `/bar`.
+ * \sa rcl_remap_topic_name()
+ * \sa rcl_remap_service_name()
+ * \sa rcl_remap_node_name()
+ * \sa rcl_remap_node_namespace()
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Yes
+ * Thread-Safe        | Yes
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param[in] argc The number of arguments in argv.
+ * \param[in] argv Whe values of the arguments.
```

Comment:
Spelling: `Whe`

Reply:
Fixed in 74b042b53f01811c99588e3a2522c35a812ce6ce